Port mozilla/amo-loadtest to FXA authentication and WebExtensions.

EnTeQuAk opened the pull request:
This is the first part of many to improve our load-tests and allow them
to be run regularly as part of a regression test suite.

This commit primarily ports the existing tests over to our repository,
updates them to use FxA authentication and uses a WebExtension for the
upload test.

This only implements the baseline for more work to come in #7744 but
it's important to have authentication and the basic infrastructure
working correctly.

It's also still using the legacy add-on detail and listing pages,
that'll obviously change - or be only true for SeaMonkey and Thunderbird
related tests since we still have to support them (very low priority though)

It also adds a first step of a summary report that links to new-relic.
That'll need a bit more tooling and testing but it worked quite well in
first tests. I also wanted to experiment with using datadog since it does support some interesting compare graphs: https://www.datadoghq.com/blog/metrics-better-or-worse-than-before/

Refs #7744

Next steps:
 * Implement most of the read-only tests from #7744
 * Check what needs auth in #7744, implement it properly
 * Implement database and cache query logging
 * Implement a unified merged test-report that uses the database and
 cache query logging and merges it with locust logs (using our new
 unique request id)

EnTeQuAk commented on requirements/perftests.txt:
```diff
@@ -0,0 +1,4 @@
+cssselect >= 0.9.1
+lxml >= 3.5.0
+pyzmq >= 15.1.0
+-e git+https://github.com/locustio/locust@524ab5203ebc7c4c5c108b641773262ae53fbc40#egg=locust
```
These will obviously be pinned later, currently I rely on that commit because it improves the event marker handling.

EnTeQuAk commented on src/olympia/wsgi.py:
```diff
@@ -44,8 +44,9 @@ def application(env, start_response):
+newrelic_uses_environment = os.environ.get('NEW_RELIC_LICENSE_KEY', None)
 
-if newrelic_ini:
+if newrelic_ini or newrelic_uses_environment:
```
Prepares for using environment variables directly to configure NewRelic, this can be used in CircleCI or Travis later to configure NewRelic and generate reports that are easy to compare.

EnTeQuAk commented:
The ui-tests are failing because of changed content on addons-frontend, I can fix them later or @jrbenny35 can look into those failures.

diox commented on requirements/perftests.txt:
```diff
@@ -0,0 +1,4 @@
+cssselect >= 0.9.1
```
hashes plz

diox commented on tests/performance/generate-summary.py:
```diff
@@ -0,0 +1,174 @@
+            account_name='Mozilla_25',
+            account_id='1402187',
+            app_name='AMO Loadtests',
+            app_id='120750863',
```
hardcoded ids ?

diox commented on tests/performance/locustfile.py:
```diff
@@ -0,0 +1,205 @@
+
+        log.debug('calling login/start to generate fxa_state')
+        response = self.client.get(
+            '/api/v3/accounts/login/start/',
```
`reverse()` all URLs?

EnTeQuAk commented:
@diox I removed the summary script for now until we have the integration working correctly. I saved it at https://gist.github.com/EnTeQuAk/9b02c27f043abb470d5d4273bbd2ed51 for now.

Also fixed the other comments.
